Content.Server/Xenoarchaeology/Artifact/XenoArtifactSystem.ProcGen.cs

```diff
@@ -0,0 +1,176 @@
+using System.Linq;
+using Content.Shared.Random.Helpers;
+using Content.Shared.Xenoarchaeology.Artifact.Components;
+using Content.Shared.Xenoarchaeology.Artifact.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.Artifact;
+
+public sealed partial class XenoArtifactSystem
+{
+    private readonly List<XenoArchTriggerPrototype> _triggerPool = new();
```

Comment:
having state in system is considered 'bad practice' need to move it into arguments